libs/sdk-ui-kit/src/Dialog/ConversionTouchPointDialogBase.tsx

```diff
@@ -0,0 +1,189 @@
+    dialogTitle?: string;
+
+    /**
+     * The value for Cancel button
+     */
+    cancelButtonText?: string;
+
+    /**
+     * Show/Hide Cancel button
+     */
+    showCancelButton?: boolean;
+
+    /**
+     * Populate values for the Hubspot form
+     */
+    values?: IHubspotFormValue;
+
+    /**
+     * Mark the checkbox as checked in the Hubspot form base on this value.
+     */
+    selectedValue?: string;
+
+    /**
+     * The id for html element render the Hubspot form
+     */
+    targetId?: string;
+
+    /**
+     * Close dialog action
+     */
+    onClose: () => void;
+
+    /**
+     * Form submitted callback function
+     */
+    onFormSubmitted?: () => void;
+}
+
+/**
+ * @internal
+ */
+export const ConversionTouchPointDialogBase: React.FC<IConversionTouchPointDialogBaseProps> = (props) => {
+    const {
+        targetId,
+        hubspotPortalId,
+        hubspotFormId,
+        selectedValue,
+        cancelButtonText,
+        dialogTitle,
+        showCancelButton,
+        values = {},
+        onClose,
+        onFormSubmitted,
+    } = props;
+
+    const intl = useIntl();
+    const [isFormReady, setIsFormReady] = useState(false);
+    const hubspotFormTargetId = targetId || "conversion-touch-point-hubspot";
+
+    useEffect(() => {
+        // the Hubspot form requires jQuery object, we need to create a fake jQuery object to bypass this.
+        window.jQuery =
+            window.jQuery ||
+            function (nodeOrSelector: string | object) {
+                if (typeof nodeOrSelector == "string") {
+                    return document.querySelector(nodeOrSelector);
+                }
+                return nodeOrSelector;
+            };
+        if (window.hbspt?.isSuccessMessageShow) {
+            window.hbspt.isSuccessMessageShow = false;
+        }
+    }, []);
+
+    const onHubspotFormSubmitted = () => {
+        if (!window.hbspt.isSuccessMessageShow) {
+            onFormSubmitted?.();
+            onClose();
+            window.hbspt.isSuccessMessageShow = true;
+        }
+    };
+
+    const { formCreated } = useHubspotForm({
+        portalId: hubspotPortalId,
+        formId: hubspotFormId,
+        target: `#${hubspotFormTargetId}`,
+        locale: intl.locale.split("-").shift() as HubSpotFormLocale,
+        onFormSubmitted: onHubspotFormSubmitted,
+        onFormReady: ($form: IHubspotFormField[]) => {
+            setIsFormReady(true);
+            // populating the values for hidden fields
+            for (let i = 0; i <= $form.length; i += 1) {
+                const inputField = $form[i];
+                if (!inputField) {
+                    return;
+                }
+
+                const populatedValue = values[inputField.name];
+
+                if (populatedValue) {
+                    inputField.value = populatedValue;
+                }
+
+                if (selectedValue && inputField.value.indexOf(selectedValue) !== -1) {
+                    setTimeout(() => {
```

Comment:
Could you please add a comment explaining why we have to call `inputField.click();` in a `setTimeout`?

Reply:
We can remove the setTimeout, it doesn't need anymore. I have tested and the form still work as expected.